### PR TITLE
SRCH-2053 remove tweetstream

### PR DESCRIPTION
**Pull request by jmax-fearless**
SRCH-2053

### CODE REVIEW

**Comment by jmax-fearless** [comment]
I am requesting approval with one outstanding CodeClimate complaint about using a before(:all) block. This is a case where that really is necessary.

**Comment by MothOnMars** [comment]
@jmax-fearless , please resolve the test coverage failure:
```
=========== Lines missing coverage: ===========
lib/importers/twitter_data.rb 51, 52
lib/importers/twitter_stream_consumer.rb 77, 78
```

**Comment by MothOnMars** [comment]
P.S. `lib/tasks/twitter.rake ` also needs to be cleaned up

**Comment by MothOnMars** [comment]
I've added a `do not merge` label to this PR, as it needs to be released in tandem with https://github.com/GSA/usasearch-cookbooks/pull/521.

File: lib/twitter_streaming_monitor.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    return if monitor&.alive?
+
+    twitter_ids = SynchronizedObjectHolder.new { TwitterProfile.active_twitter_ids }
+    self.monitor = TwitterStreamingMonitor.new(twitter_ids)
+    monitor.run
+  end
+
+  def self.stop
+    monitor.stop
+  end
```
Comment (MothOnMars) [inline lib/twitter_streaming_monitor.rb:32]: per our discussion, refactor to remove redundant class methods

**Review by MothOnMars — CHANGES_REQUESTED** [review]
Per our discussion, some refactoring is needed:
- review misc. notes
- prevent the rake task from dying silently
- remove class methods from `lib/twitter_streaming_monitor.rb`
- add unit tests for `lib/twitter_streaming_monitor.rb` and `lib/importers/twitter_stream_consumer.rb` (both classes should be 100% covered without running the rake task specs)
- once we have unit tests on the above, we can remove the pseudo-integration specs for the rake task, as they aren't really functioning as end-to-end tests without our being able to stub or record the Twitter responses

File: spec/lib/tasks/twitter_spec.rb [hunk 2]
```diff
@@ -1,35 +1,50 @@
...
-  before(:all) do
-    @rake = Rake::Application.new
-    Rake.application = @rake
+  before :all do
+    Rake.application = Rake::Application.new
     Rake.application.rake_require('tasks/twitter')
     Rake::Task.define_task(:environment)
   end
 
+  let(:task_name) { nil }
```
[the remark this answers is not in the document]
Comment (jmax-fearless) [reply spec/lib/tasks/twitter_spec.rb:23]: Removed.

File: spec/lib/tasks/twitter_spec.rb [hunk 3]
```diff
@@ -41,251 +56,286 @@
...
+            expect(Rails.logger).to have_received(:error).
+              with(/\[TWITTER\] \[FOLLOW\] \[ERROR\].*error while deleting tweet#1234/).
+              at_least(:once)
           end
         end
       end
     end
   end
 end
+j
```
[the remark this answers is not in the document]
Comment (jmax-fearless) [reply spec/lib/tasks/twitter_spec.rb]: Gone.

File: lib/twitter_streaming_monitor.rb (hunk 1 above)
Comment (jmax-fearless) [reply lib/twitter_streaming_monitor.rb]: Removed.

File: spec/lib/tasks/twitter_spec.rb [hunk 4]
```diff
@@ -41,251 +56,286 @@
...
+          it 'saves the tweet text' do
+            expect(tweet.tweet_text).to eq('Fast. Relevant. Free. Features: http://t.co/l8VhWiZH http://t.co/y5YSDq7M')
+          end
+
+          it 'saves the tweet urls' do
+            expect(tweet.urls.collect(&:display_url)).to eq(%w[search.gov/features pic.twitter.com/y5YSDq7M])
+          end
         end
 
-        it 'should persist urls with complete data' do
```
[the remark this answers is not in the document]
Comment (jmax-fearless) [reply spec/lib/tasks/twitter_spec.rb:180]: Moved to twitter_stream_consumer_spec.rb_

**Comment by MothOnMars** [comment]
Closing this up in favor of a new approach.

**Pull request closed without merging**